dali/kernels/imgproc/convolution/separable_convolution_cpu.h

```diff
@@ -104,15 +104,17 @@ struct SeparableConvolutionCpu<Out, In, W, 2, has_channels> {
            const TensorView<StorageCPU, const In, ndim>& in,
            const std::array<TensorView<StorageCPU, const W, 1>, axes>& windows,
            W scale = 1) {
-    auto *tmp = ctx.scratchpad->Allocate<Intermediate>(AllocType::Host, volume(in.shape));
+    auto *tmp = ctx.scratchpad->AllocateHost<Intermediate>(volume(in.shape));
     auto intermediate = TensorView<StorageCPU, Intermediate, ndim>(tmp, in.shape);
 
     // Prepare the scratchpad with all the remaining memory requested by sub-kernels
+    // TODO(michalz): Get rid of this run-time memory kind dispatch
     PreallocatedScratchpad sub_scratch;
     for (size_t i = 0; i < sub_scratch_sizes_.size(); i++) {
       auto sz = sub_scratch_sizes_[i];
-      auto alloc_type = static_cast<AllocType>(i);
-      sub_scratch.allocs[i] = BumpAllocator(ctx.scratchpad->Allocate<char>(alloc_type, sz, 64), sz);
+      auto kind_id = static_cast<mm::memory_kind_id>(i);
```

Comment:
So each element of `sub_scratch_sizes_` corresponds to a different memory type?

Reply:
Just like it did before.

Reply:
Just asking.

Reply:
Can't you unroll like in `normalize_gpu.cu`?

Reply:
In principle, I could - but would it be really be any better?